File: frontend/app/controller/model/BureauOrdre/CourrielHistory.ts
import { UtilisateurDto } from "app/controller/model/Utilisateur.model";
import { CourrielBureauOrdre } from "./CourrielBureauOrdre";
import { IntervenantCourriel } from "./IntervenantCourriel";

export class CourrielHistory {
    id: number;
    message: string;
    type: string;
    dateAction: Date;

    initiator: UtilisateurDto;
    courriel?: CourrielBureauOrdre;
    intervenant: IntervenantCourriel;

    constructor() {
        this.id = 0;
        this.message = '';
        this.type = '';
        this.dateAction = new Date();

        this.initiator = new UtilisateurDto();
        this.courriel = new CourrielBureauOrdre();
        this.intervenant = new IntervenantCourriel();
    }
}

File: backend-ms2/src/main/java/ma/sir/ged/bean/core/bureauOrdre/History/CourrielHistory.java
package ma.sir.ged.bean.core.bureauOrdre.History;

import lombok.Data;
import ma.sir.ged.bean.core.bureauOrdre.CourrielBo;
import ma.sir.ged.bean.core.bureauOrdre.IntervenantsCourriel;
import ma.sir.ged.bean.core.bureauOrdre.enums.HistoryEntryType;
import ma.sir.ged.bean.core.organigramme.Utilisateur;
import java.time.LocalDateTime;

import javax.persistence.*;

@Entity
@Data
public class CourrielHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String message;

    private HistoryEntryType type;

    @Column(columnDefinition = "DATETIME")
    private LocalDateTime dateAction;

    @ManyToOne
    private Utilisateur initiator;

    @ManyToOne
    private CourrielBo courriel;

    @OneToOne
    private IntervenantsCourriel intervenant;
}
